x-pack/plugins/cross_cluster_replication/__jest__/client_integration/auto_follow_pattern_list.test.js

```diff
@@ -0,0 +1,323 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import sinon from 'sinon';
+
+import { initTestBed, mockServerResponses, nextTick, getRandomString, findTestSubject } from './test_helpers';
+import { AutoFollowPatternList } from '../../public/app/sections/home/auto_follow_pattern_list';
+import { getAutoFollowPatternClientMock } from '../../fixtures/auto_follow_pattern';
+
+jest.mock('ui/chrome', () => ({
+  addBasePath: () => 'api/cross_cluster_replication',
+  breadcrumbs: { set: () => {} },
+}));
+
+jest.mock('ui/index_patterns', () => {
+  const { INDEX_PATTERN_ILLEGAL_CHARACTERS_VISIBLE } = require.requireActual('../../../../../src/legacy/ui/public/index_patterns/constants'); // eslint-disable-line max-len
+  return { INDEX_PATTERN_ILLEGAL_CHARACTERS_VISIBLE };
+});
+
+describe('<AutoFollowPatternList />', () => {
+  let server;
+  let find;
+  let exists;
+  let component;
+  let getMetadataFromEuiTable;
+  let getUserActions;
+  let tableCellsValues;
+  let rows;
+  let mockLoadAutoFollowPattern;
+  let mockDeleteAutoFollowPattern;
+  let mockAutoFollowStats;
+
+  beforeEach(() => {
+    server = sinon.fakeServer.create();
+    server.respondImmediately = true;
+    ({ mockLoadAutoFollowPattern, mockDeleteAutoFollowPattern, mockAutoFollowStats } = mockServerResponses(server));
```

Comment:
It looks like each of these helpers is only used once in this file. Instead of defining them in a `beforeEach`, can we define them only within the test? Can we also declare them within those tests, to scope them more closely to where they're used? If you agree that this will help make it clearer where these helpers are being used, then I'd suggest applying this idea elsewhere too, if possible (I haven't had time to check yet).

Reply:
As from my comments above, this would make things much more complex than needed. To keep simplicity 2 things happen:

- Mock all server requests to a default value (once on before each at the top)
- _only_ when needed, override a request, close to the test, for special use cases